mlrun/datastore/sources.py

```diff
@@ -200,15 +206,18 @@ def to_spark_df(self, session, named_view=False, time_field=None):
                 df = df.withColumn(col_name, funcs.col(col_name).cast("timestamp"))
         if named_view:
             df.createOrReplaceTempView(self.name)
-        return df
+        return self._filter_spark_df(df, time_field, columns)
 
-    def to_dataframe(self):
-        kwargs = self.attributes.get("reader_args", {})
+    def to_dataframe(self, **kwargs):
```

Comment:
Here too, better list the parameters we expect, i.e. `parse_dates` and `chunksize`. And if we need a passthrough `reader_args` (do we?), it's probably best to make that a regular parameter too, instead of mixing it up with `kwargs`.